src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/UnitTests/Controllers/AccountControllerTests.cs

```diff
@@ -0,0 +1,774 @@
+            var viewResult = Assert.IsType<RedirectToActionResult>(result.Result);
+            Assert.Equal("ForgotPasswordConfirmation", viewResult.ActionName);
+        }
+
+        [Fact]
+        public void ForgotPassword_UserIsCorrect_RedirectToAction()
+        {
+            // Arrange
+            var user = new ApplicationUser();
+            var model = new ForgotPasswordViewModel();
+            model.Email = "test@gmail.com";
+            this._accountServiceMock.Setup(x => x.FindByEmailAsync(model.Email)).ReturnsAsync(user);
+            this._accountServiceMock.Setup(x => x.IsEmailConfirmedAsync(user)).ReturnsAsync(true);
+            using var controller = new AccountController(this._accountServiceMock.Object, this._loggerMock.Object);
+
+            // Act
+            var result = controller.ForgotPassword(model);
+
+            // Arrange - *********************zrobic*********************
+        }
+
+        [Fact]
+        public void ForgotPasswordConfirmation_ReturnsViewResult()
+        {
+            // Arrange
+            using var controller = new AccountController(this._accountServiceMock.Object, this._loggerMock.Object);
+
+            // Act
+            var result = controller.ForgotPasswordConfirmation();
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.Null(viewResult.Model);
+        }
+
+        [Fact]
+        public void ResetPassword_CodeIsNull_ThrowsException()
+        {
+            // Arrange
+            var email = "test@gmail.com";
+            string code = null;
+            using var controller = new AccountController(this._accountServiceMock.Object, this._loggerMock.Object);
+
+            // Act
+            Func<IActionResult> act = () => controller.ResetPassword(email, code);
+
+            // Assert
+            Assert.Throws<ArgumentException>(act);
+        }
+
+        [Fact]
+        public void ResetPassword_CodeIsNotNull_ReturnsViewResult()
+        {
+            // Arrange
+            var email = "test@gmail.com";
+            string code = "code";
+            using var controller = new AccountController(this._accountServiceMock.Object, this._loggerMock.Object);
+
+            // Act
+            var result = controller.ResetPassword(email, code);
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.IsType<ResetPasswordViewModel>(viewResult.ViewData.Model);
+        }
+
+        [Fact]
+        public void ResetPassword_ResetPasswordViewModelIsNull_ThrowsException()
+        {
+            // Arrange
+            ResetPasswordViewModel model = null;
+            using var controller = new AccountController(this._accountServiceMock.Object, this._loggerMock.Object);
+
+            // Act
+            Func<IActionResult> act = () => controller.ResetPassword(model).Result;
+
+            // Assert
+            Assert.Throws<AggregateException>(act);
```

Comment:
The actual exception is ArgumentNullException. Using async/await avoids the AggregateException wrapper if the task has an exception. The code could look like this
`Func<Task<IActionResult>> act = async () => await controller.ResetPassword(model).ConfigureAwait(false); `
`await Assert.ThrowsAsync<ArgumentNullException>(act).ConfigureAwait(false);`

Reply:
He I did changes as you suggested

![obraz](https://user-images.githubusercontent.com/27915290/98267943-3b8e5c00-1f8c-11eb-928a-27f74f16a2e5.png)

but there is stm wrong as test passed with different kind of errors eg:

![obraz](https://user-images.githubusercontent.com/27915290/98268173-7e503400-1f8c-11eb-98ad-90036b761018.png)

etc.

Reply:
In this case you need to also await that assertion

Reply:
Thanks!